### PR TITLE
add Formula1800

### PR DESCRIPTION
**Pull request by swiftrax**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kikoslab/formula1800/info.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+    "keyboard_name": "formula1800", 
+    "url": "https://github.com/swiftrax", 
+    "maintainer": "swiftrax", 
+    "layouts": {
+        "LAYOUT": {
```
Comment (zvecr) [inline keyboards/kikoslab/formula1800/info.json:6]:
```suggestion
        "LAYOUT_all": {
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/kikoslab/formula1800/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        }
+    } else if (index == 1) { /* Second encoder */
+        if (clockwise) {
+            tap_code(KC_VOLD);
+        } else {
+            tap_code(KC_VOLU);
+        }
+    }
+    return true;
+}
```
Comment (drashna) [inline keyboards/kikoslab/formula1800/keymaps/default/keymap.c:62]:
```suggestion
```

File: keyboards/kikoslab/formula1800/formula1800.c [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include "formula1800.h"
```
Comment (drashna) [inline keyboards/kikoslab/formula1800/formula1800.c:17]: This allows for a default behavior to be defined, without the need for the user to specify the behavior.  This is very useful for the configurator which doesn't have a way to configure this behavior, currently.

```suggestion
#include "formula1800.h"

#ifdef ENCODER_ENABLE
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
    if (index == 0) { /* First encoder */
        if (clockwise) {
            tap_code(KC_WH_D);
        } else {
            tap_code(KC_WH_U);
        }
    } else if (index == 1) { /* Second encoder */
        if (clockwise) {
            tap_code_delay(KC_VOLD, 10);
        } else {
            tap_code_delay(KC_VOLU, 10);
        }
    }
    return true;
}
#endif
```

File: keyboards/kikoslab/formula1800/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        }
+    } else if (index == 1) { /* Second encoder */
+        if (clockwise) {
+            tap_code(KC_VOLD);
+        } else {
+            tap_code(KC_VOLU);
+        }
+    }
+    return true;
+}
```
Comment (drashna) [inline keyboards/kikoslab/formula1800/keymaps/via/keymap.c:78]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/kikoslab/formula1800/info.json [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
...25, "y":4.25, "w":1.25}, {"label":"9,2", "x":5.5, "y":4.25}, {"label":"8,3", "x":6.5, "y":4.25}, {"label":"9,3", "x":7.5, "y":4.25}, {"label":"8,4", "x":8.5, "y":4.25}, {"label":"9,4", "x":9.5, "y":4.25}, {"label":"8,5", "x":10.5, "y":4.25}, {"label":"9,5", "x":11.5, "y":4.25}, {"label":"8,6", "x":12.5, "y":4.25}, {"label":"9,6", "x":13.5, "y":4.25}, {"label":"8,7", "x":14.5, "y":4.25}, {"label":"9,7", "x":15.5, "y":4.25}, {"label":"8,8", "x":16.5, "y":4.25, "w":1.75}, {"label":"9,9", "x":19.75, "y":4.25}, {"label":"9,8", "x":18.5, "y":4.5}, {"label":"10,0", "x":0, "y":5.25}, {"label":"11,0", "x":1, "y":5.25}, {"label":"10,1", "x":2, "y":5.25}, {"label":"11,1", "x":3, "y":5.25}, {"label":"10,2", "x":4.25, "y":5.25, "w":1.25}, {"label":"11,2", "x":5.5, "y":5.25, "w":1.25}, {"label":"10,3", "x":6.75, "y":5.25, "w":1.25}, {"label":"11,4", "x":8, "y":5.25, "w":2.25}, {"label":"10,5", "x":10.25, "y":5.25, "w":1.25}, {"label":"11,6", "x":11.5, "y":5.25, "w":2.75}, {"label":"10,7", "x":14.25, "y":5.25}, {"label":"11,7", "x":15.25, "y":5.25}, {"label":"10,8", "x":16.25, "y":5.25}, {"label":"11,8", "x":17.5, "y":5.5}, {"label":"10,9", "x":18.5, "y":5.5}, {"label":"11,9", "x":19.5, "y":5.5}]
```
Comment (noroadsleft) [inline keyboards/kikoslab/formula1800/info.json:7]: Fixes an error in the key sequence.

```suggestion
            "layout": [{"label":"0,0", "x":0, "y":0}, {"label":"1,0", "x":1, "y":0}, {"label":"0,1", "x":2, "y":0}, {"label":"1,1", "x":3, "y":0}, {"label":"0,2", "x":4.25, "y":0}, {"label":"1,2", "x":5.5, "y":0}, {"label":"0,3", "x":6.5, "y":0}, {"label":"1,3", "x":7.5, "y":0}, {"label":"0,4", "x":8.5, "y":0}, {"label":"1,4", "x":9.75, "y":0}, {"label":"0,5", "x":10.75, "y":0}, {"label":"1,5", "x":11.75, "y":0}, {"label":"0,6", "x":12.75, "y":0}, {"label":"1,6", "x":14, "y":0}, {"label":"0,7", "x":15, "y":0}, {"label":"1,7", "x":16, "y":0}, {"label":"0,8", "x":17, "y":0}, {"label":"0,9", "x":18.25, "y":0}, {"label":"1,9", "x":19.75, "y":0}, {"label":"2,0", "x":0, "y":1.25}, {"label":"3,0", "x":1, "y":1.25}, {"label":"2,1", "x":2, "y":1.25}, {"label":"3,1", "x":3, "y":1.25}, {"label":"2,2", "x":4.25, "y":1.25}, {"label":"3,2", "x":5.25, "y":1.25}, {"label":"2,3", "x":6.25, "y":1.25}, {"label":"3,3", "x":7.25, "y":1.25}, {"label":"2,4", "x":8.25, "y":1.25}, {"label":"3,4", "x":9.25, "y":1.25}, {"label":"2,5", "x":10.25, "y":1.25}, {"label":"3,5", "x":11.25, "y":1.25}, {"label":"2,6", "x":12.25, "y":1.25}, {"label":"3,6", "x":13.25, "y":1.25}, {"label":"2,7", "x":14.25, "y":1.25}, {"label":"3,7", "x":15.25, "y":1.25}, {"label":"2,8", "x":16.25, "y":1.25}, {"label":"3,8", "x":17.25, "y":1.25}, {"label":"2,9", "x":18.25, "y":1.25}, {"label":"3,9", "x":19.75, "y":1.25}, {"label":"4,0", "x":0, "y":2.25}, {"label":"5,0", "x":1, "y":2.25}, {"label":"4,1", "x":2, "y":2.25}, {"label":"5,1", "x":3, "y":2.25}, {"label":"4,2", "x":4.25, "y":2.25, "w":1.5}, {"label":"5,2", "x":5.75, "y":2.25}, {"label":"4,3", "x":6.75, "y":2.25}, {"label":"5,3", "x":7.75, "y":2.25}, {"label":"4,4", "x":8.75, "y":2.25}, {"label":"5,4", "x":9.75, "y":2.25}, {"label":"4,5", "x":10.75, "y":2.25}, {"label":"5,5", "x":11.75, "y":2.25}, {"label":"4,6", "x":12.75, "y":2.25}, {"label":"5,6", "x":13.75, "y":2.25}, {"label":"4,7", "x":14.75, "y":2.25}, {"label":"5,7", "x":15.75, "y":2.25}, {"label":"4,8", "x":16.75, "y":2.25}, {"label":"5,8", "x":17.75, "y":2.25, "w":1.5}, {"label":"5,9", "x":19.75, "y":2.25}, {"label":"6,0", "x":0, "y":3.25}, {"label":"7,0", "x":1, "y":3.25}, {"label":"6,1", "x":2, "y":3.25}, {"label":"7,1", "x":3, "y":3.25}, {"label":"6,2", "x":4.25, "y":3.25, "w":1.75}, {"label":"7,2", "x":6, "y":3.25}, {"label":"6,3", "x":7, "y":3.25}, {"label":"7,3", "x":8, "y":3.25}, {"label":"6,4", "x":9, "y":3.25}, {"label":"7,4", "x":10, "y":3.25}, {"label":"6,5", "x":11, "y":3.25}, {"label":"7,5", "x":12, "y":3.25}, {"label":"6,6", "x":13, "y":3.25}, {"label":"7,6", "x":14, "y":3.25}, {"label":"6,7", "x":15, "y":3.25}, {"label":"7,7", "x":16, "y":3.25}, {"label":"6,8", "x":17, "y":3.25}, {"label":"7,8", "x":18, "y":3.25, "w":1.25}, {"label":"7,9", "x":19.75, "y":3.25}, {"label":"8,0", "x":0, "y":4.25}, {"label":"9,0", "x":1, "y":4.25}, {"label":"8,1", "x":2, "y":4.25}, {"label":"9,1", "x":3, "y":4.25}, {"label":"8,2", "x":4.25, "y":4.25, "w":1.25}, {"label":"9,2", "x":5.5, "y":4.25}, {"label":"8,3", "x":6.5, "y":4.25}, {"label":"9,3", "x":7.5, "y":4.25}, {"label":"8,4", "x":8.5, "y":4.25}, {"label":"9,4", "x":9.5, "y":4.25}, {"label":"8,5", "x":10.5, "y":4.25}, {"label":"9,5", "x":11.5, "y":4.25}, {"label":"8,6", "x":12.5, "y":4.25}, {"label":"9,6", "x":13.5, "y":4.25}, {"label":"8,7", "x":14.5, "y":4.25}, {"label":"9,7", "x":15.5, "y":4.25}, {"label":"8,8", "x":16.5, "y":4.25, "w":1.75}, {"label":"9,8", "x":18.5, "y":4.5}, {"label":"9,9", "x":19.75, "y":4.25}, {"label":"10,0", "x":0, "y":5.25}, {"label":"11,0", "x":1, "y":5.25}, {"label":"10,1", "x":2, "y":5.25}, {"label":"11,1", "x":3, "y":5.25}, {"label":"10,2", "x":4.25, "y":5.25, "w":1.25}, {"label":"11,2", "x":5.5, "y":5.25, "w":1.25}, {"label":"10,3", "x":6.75, "y":5.25, "w":1.25}, {"label":"11,4", "x":8, "y":5.25, "w":2.25}, {"label":"10,5", "x":10.25, "y":5.25, "w":1.25}, {"label":"11,6", "x":11.5, "y":5.25, "w":2.75}, {"label":"10,7", "x":14.25, "y":5.25}, {"label":"11,7", "x":15.25, "y":5.25}, {"label":"10,8", "x":16.25, "y":5.25}, {"label":"11,8", "x":17.5, "y":5.5}, {"label":"10,9", "x":18.5, "y":5.5}, {"label":"11,9", "x":19.5, "y":5.5}]
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Pull request closed without merging**